packages/mrm/src/index.js

```diff
@@ -143,38 +143,52 @@ function runTask(taskName, directories, options, argv) {
 			);
 		}
 
-		const resolvedConfig = argv.interactive
-			? getInteractiveConfig(module, options)
-			: options;
+		const options = processTaskOptions(module, argv.interactive, defaults);
+
+		console.log(kleur.cyan(`Running ${taskName}...`));
 
-		Promise.resolve(resolvedConfig)
+		Promise.resolve(options)
 			.then(getConfigGetter)
-			.then(config => {
-				console.log(kleur.cyan(`Running ${taskName}...`));
-				return module(config, argv);
-			})
+			.then(config => module(config, argv))
 			.catch(reject)
 			.then(resolve);
 	});
 }
 
 /**
- * Get task specific interactive config options.
+ * Get task specific interactive config options (by using prompt or defaults).
  *
  * @param {Object} task
+ * @param {Object} interactive? Whether or not interactive mode is enabled.
+ * @param {Object} options? Default available options passed into the task.
  */
-function getInteractiveConfig(task, initials = {}) {
+function processTaskOptions(task, interactive = false, options = {}) {
+	// Avoid mutation, but keep code simplicity
+	const defaults = { ...options };
+
+	// If no parameters set, resolve to default options (from config file or command line).
 	if (!task.parameters) {
-		return Promise.resolve({});
+		return Promise.resolve(options);
 	}
 
-	const prompts = Object.entries(task.parameters).map(([name, prompt]) => ({
-		...prompt,
-		name,
-		initial: initials[name] || prompt.initial,
-	}));
+	const prompts = [];
+	const parameters = Object.entries(task.parameters);
+
+	for (const [name, prompt] of parameters) {
```

Comment:
This could be `.map()`.

Reply:
This could... but I'm doing two mutations inside this loop: one to merge default values, one to prepare prompts. `map` could mislead a reader into thinking it's mutation safe, as we usually expect a `map` to return altered values but without side-effects. `forEach` would be fit for this, at least to make it well clear it's mutation based, but I see no much benefit of `forEach` over syntax features such as `for` loops.

Reply:
>  two mutations inside this loop

That's quite unfortunate ;-/ I would probably not mix two loops into one.

Something like 

```js
const initials = parameters.reduce(([name, prompt], acc) => ({
  ...acc,
  [name]: prompt.initial !== 'undefined' ? prompt.initial : defaults[name]
}), {})

const prompts = parameters.map(([name, prompt]) => ({ ...prompt, name, initial: initials[name] })
```

Reply:
I agree this can be done this way. Honestly, for open-source and community sake, I find the previous version more readable.

Ps.: just shipped a code here to fix `initial` being a function (and even async, as per enquirer docs). Any recommendation on promise awaiting inside reducers (that's the affected part)?

Reply:
It might be more readable but it has a surprising behavior ;-)

For the async I'd probably use `Promise.all` with initial value for all parameters, even sync or undefined, so you could match them to respective parameters.

Reply:
You wouldn't need to disable `no-await-in-loop` with this approach ;-)